src/expr/src/expr/expr_split_part.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use risingwave_common::array::{
+    Array, ArrayBuilder, ArrayImpl, ArrayRef, DataChunk, Utf8ArrayBuilder,
+};
+use risingwave_common::error::{Result, RwError};
+use risingwave_common::types::DataType;
+use risingwave_common::{ensure, try_match_expand};
+use risingwave_pb::expr::expr_node::{RexNode, Type};
+use risingwave_pb::expr::ExprNode;
+
+use crate::expr::{build_from_prost, BoxedExpression, Expression};
+
+#[derive(Debug)]
+pub struct SplitPartExpression {
+    return_type: DataType,
+    string_expr: BoxedExpression,
+    delimiter_expr: BoxedExpression,
+    nth_expr: BoxedExpression,
+}
+
+impl Expression for SplitPartExpression {
```

Comment:
Avoid `impl Expression` directly when possible. Take a look at `TernaryBytesExpression`.

Reply:
thx, fixed

is [455d9d0](https://github.com/singularity-data/risingwave/pull/2822/commits/455d9d0aa399079e9fc61d1d6ddd08c9eaaf7fe1) okay to return error code here?